test/Benchmarks/OrleansBenchmarks/MapReduce/MapReduceBenchmark.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using Orleans.TestingHost;
+using OrleansBenchmarkGrains.MapReduce;
+using OrleansGrainInterfaces.MapReduce;
+
+namespace OrleansBenchmarks.MapReduce
+{
+    public class MapReduceBenchmark
+    {
+        private static SiloHandle _siloHandle;
+        private readonly int _intermediateStagesCount = 15;
+        private readonly int _pipelineParallelization = 4;
+        private readonly int _repeats = 50000;
+        private int _currentRepeat = 0;
+
+        [Setup]
+        public void BenchmarkSetup()
+        {
+            var nodeConfig = new NodeConfiguration
+            {
+                TraceToConsole = false,
+                DefaultTraceLevel = Severity.Warning
+            };
+
+            var clusterConfig = ClusterConfiguration.LocalhostPrimarySilo();
+            clusterConfig.Defaults.TraceToConsole = false;
+            _siloHandle = TestCluster.StartOrleansSilo(
+                new TestCluster(),
+                Silo.SiloType.Primary,
+                clusterConfig,
+                nodeConfig);
+
+            var clientConfig = ClientConfiguration.LocalhostSilo();
+            clientConfig.TraceToConsole = false;
+            GrainClient.Initialize(clientConfig);
```

Comment:
you should be able to replace this whole set up with the equivalent higher level set up:
```
            TestClusterOptions.DefaultTraceToConsole = false;
            var options = new TestClusterOptions(1);
            options.ClusterConfiguration.ApplyToAllNodes(c => c.DefaultTraceLevel = Severity.Warning);
            _host = new TestCluster(options);
            _host.Deploy();
```
This has a few additional benefits (such as allowing to start more than 1 silo and it being configured automatically, using random ports for networking, and a little more future proof if we have to use some external process manager when we move to .net core).

Also, if there is a concept of tear down in the benchmark framework, please consider calling `_host.StopAllSilos();` to avoid this cluster leaking into the next benchmark being run.